Firestore/core/src/core/firestore_client.cc

```diff
@@ -145,10 +145,19 @@ std::shared_ptr<FirestoreClient> FirestoreClient::Create(
   };
 
   shared_client->app_check_credentials_provider_->SetCredentialChangeListener(
-      [](std::string) {
-        // Register an empty credentials change listener to activate token
-        // refresh.
+      [weak_client](const std::string&) {
+        auto shared_client = weak_client.lock();
+        if (!shared_client) return;
+        if (shared_client->credentials_initialized_) {
```

Comment:
Can you add a comment here why it is okay to swallow the notification if we have not yet received an auth credential? I think it is okay since we will always fetch a token during client startup, so even if we ignore a token once here we would receive it at a later point.

Reply:
That's right. Done.